### ISSUE TITLE
php error call_user_func_array() expects parameter 1 to be a valid callback

### ISSUE THREAD

**Issue by kghbln** [SemanticMediaWiki/SemanticFormsSelect#4]
> Reported [on mw.o](https://www.mediawiki.org/wiki/Topic:Sw9visnfq4ghcrtv)

I'm trying to implement a SF_Select field. The field definition does not appear to get parsed; when I open the form with the field definition the field definition is shown instead of the field itself.

I have found an apache error message which is likely to be relevant. It's an error pointing to a SemanticForms php file, but this error is only triggered when I open a form which has the SF_Select field on it. The full error is as follows:

[Mon Jan 11 15:30:06.897122 2016] [:error] [pid 21078] [client 176.20.101.243:64678] FastCGI: server "/usr/lib/cgi-bin/php5-fcgi" stderr: PHP message: PHP Warning: call_user_func_array() expects parameter 1 to be a valid callback, class '\SFS\SemanticFormsSelect' not found in /var/www/wiki/extensions/SemanticForms/includes/SF_FormPrinter.php on line 1955

My versions are as follows:
- MW: 1.24.4
- SMW: 2.3
- SF: 3.2
- SFS: 1.3 (assuming)

**Comment by mwjames** [comment]
>  PHP message: PHP Warning: call_user_func_array() expects parameter 1 to be a valid callback, class '\SFS\SemanticFormsSelect' not found in /var/www/wiki/extensions/SemanticForms/includes/SF_FormPrinter.php on line 1955

This sounds like the installation is missing something, s(he) should try to use the sandbox.

**Comment by JosefAssad** [comment]
I'm the reporter; thanks for posting this issue @kghbln. The SFS version is indeed 1.3.0 (081579f).

Any other things to install I might have missed?

**Comment by mwjames** [comment]
> Any other things to install I might have missed?

If you followed the instructions, no.

Since [0] is working as expected, I'm not sure why `class '\SFS\SemanticFormsSelect' not found` appears unless your setup cannot find the autoloader (class autoloader managed by Composer) because of some non-Composer usage or having manually copied files. The file in question is present [1] and registered upon using Composer automatically.

[0] http://sandbox.semantic-mediawiki.org/wiki/Semantic_Forms_Select_demo
[1] https://github.com/SemanticMediaWiki/SemanticFormsSelect/blob/master/src/SemanticFormsSelect.php

**Comment by JosefAssad** [comment]
Hm. OK, I got a clue as to where the problem might be from your comment @mwjames.

The installation instructions I followed are the ones on mediawiki.org [0]. Since these instructions do not mention any specific way of obtaining the extension, I git cloned it and added `require_once "$IP/extensions/SemanticFormsSelect/SemanticFormsSelect.php";` to `LocalSettings.php`.

It sounds like I should use Composer instead to install it?

[0] https://www.mediawiki.org/wiki/Extension:Semantic_Forms_Select#Installation
